core/transcoder/codecs.go

```diff
@@ -381,6 +382,64 @@ func (c *Video4Linux) GetPresetForLevel(l int) string {
 	return presetMapping[l]
 }
 
+// VideoToolboxCodec represents an instance of the VideoToolbox codec.
+type VideoToolboxCodec struct {
+}
+
+// Name returns the codec name.
+func (c *VideoToolboxCodec) Name() string {
+	return "h264_videotoolbox"
+}
+
+// DisplayName returns the human readable name of the codec.
+func (c *VideoToolboxCodec) DisplayName() string {
+	return "VideoToolbox"
+}
+
+// GlobalFlags are the global flags used with this codec in the transcoder.
+func (c *VideoToolboxCodec) GlobalFlags() string {
+	var flags []string
+
+	return strings.Join(flags, " ")
+}
+
+// PixelFormat is the pixel format required for this codec.
+func (c *VideoToolboxCodec) PixelFormat() string {
+	return "nv12"
+}
+
+// ExtraFilters are the extra filters required for this codec in the transcoder.
+func (c *VideoToolboxCodec) ExtraFilters() string {
+	return ""
+}
+
+// ExtraArguments are the extra arguments used with this codec in the transcoder.
+func (c *VideoToolboxCodec) ExtraArguments() string {
+	return ""
+}
+
+// VariantFlags returns a string representing a single variant processed by this codec.
+func (c *VideoToolboxCodec) VariantFlags(v *HLSVariant) string {
+	return ""
+}
+
+// GetPresetForLevel returns the string preset for this codec given an integer level.
+func (c *VideoToolboxCodec) GetPresetForLevel(l int) string {
+	presetMapping := []string{
+		"ultrafast -realtime true",
+		"superfast -realtime true",
+		"veryfast -realtime true",
```

Comment:
Are the presets actually called `ultrafast -realtime true`?

Reply:
You're right, I might as well do it correctly.